### DISCUSSION TITLE
Test against Redisson

### DISCUSSION THREAD

**Discussion by tisonkun** [apache/kvrocks#1912] [General]
Redis clients' test suites can always help us to verify Kvrocks' quality.

I'm starting the regression test with Redisson and this thread is for discussion and see if we can do any further work into our codebase or theirs.

For running test:

1. Clone and build Kvrocks.
2. Clone redisson at https://github.com/redisson/redisson
3. Edit [`redisBinaryPath`](https://github.com/redisson/redisson/blob/4ad13b9cc87fa6a3c562d0c33072bc300eefa26b/redisson/src/test/java/org/redisson/RedissonRuntimeEnvironment.java#L12) to the kvrocks' bin path (see also https://github.com/redisson/redisson/issues/5456)
4. Run `mvn -Punit-test clean test -e ` from Redisson's project base dir.

Still, I get a lot of error reports, so it's not smooth to integrate with our CI workflows (this is why I created a discussion instead of a tracking issue). But the result can be a hint.

**Comment by jihuayu** [comment]
Similarly, we can also run kvrocks' test cases on Redis.
It can help us to verify Kvrocks' test cases quality.
